Remove the CSSIndividualTransformPropertiesEnabled flag

graouts opened the pull request:
#### 3635c66e7e2a1fce3be707e83bae328631db3a33
<pre>
Remove the CSSIndividualTransformPropertiesEnabled flag
<a href="https://bugs.webkit.org/show_bug.cgi?id=217899">https://bugs.webkit.org/show_bug.cgi?id=217899</a>
<a href="https://rdar.apple.com/70434096">rdar://70434096</a>

Reviewed by Dean Jackson.

This flag has been on by default for years and is ripe for removal.

* LayoutTests/fast/css/rotate-invalidate-if-disabled-expected.txt: Removed.
* LayoutTests/fast/css/rotate-invalidate-if-disabled.html: Removed.
* LayoutTests/fast/css/scale-invalidate-if-disabled-expected.txt: Removed.
* LayoutTests/fast/css/scale-invalidate-if-disabled.html: Removed.
* LayoutTests/fast/css/translate-invalidate-if-disabled-expected.txt: Removed.
* LayoutTests/fast/css/translate-invalidate-if-disabled.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::CSSParserContext):
* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences CSSIndividualTransformPropertiesEnabled]): Deleted.
(-[WebPreferences setCSSIndividualTransformPropertiesEnabled:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/271133@main">https://commits.webkit.org/271133@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33ace216ec7a63b09397501fcb5d429a50ba729e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27459 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6101 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28710 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29688 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25138 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27930 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8016 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3496 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/24933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27724 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/4890 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/23577 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/4260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/4416 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/24573 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/30324 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/23889 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/25088 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/24996 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/30551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/26638 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4437 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/2573 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/28519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5903 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/34094 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4894 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/7384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3546 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4824 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
```diff
@@ -3066,16 +3066,6 @@ - (void)setMediaRecorderEnabled:(BOOL)flag
-{
-    [self _setBoolValue:flag forKey:WebKitCSSIndividualTransformPropertiesEnabledPreferenceKey];
-}
-
```
You have to noop these and move them to the end of the file. Unless you want to deal with binary compatibility fallout as I understand it.

nt1m replied:
See https://github.com/WebKit/WebKit/pull/20896 as an example of how bincompat can be preserved.

grorg replied:
I guess you could do a codesearch to see if anyone used them. I doubt it in this case, but you never know.

graouts replied:
Do we have a policy for the removal of such private preferences provided we don't find any internal clients (as is the case here)?

annevk reviewed (commented):
Non-reviewer looks good modulo binary compatibility.

graouts committed:
Remove the CSSIndividualTransformPropertiesEnabled flag
https://bugs.webkit.org/show_bug.cgi?id=217899
rdar://70434096

Reviewed by Dean Jackson.

This flag has been on by default for years and is ripe for removal.

* LayoutTests/fast/css/rotate-invalidate-if-disabled-expected.txt: Removed.
* LayoutTests/fast/css/rotate-invalidate-if-disabled.html: Removed.
* LayoutTests/fast/css/scale-invalidate-if-disabled-expected.txt: Removed.
* LayoutTests/fast/css/scale-invalidate-if-disabled.html: Removed.
* LayoutTests/fast/css/translate-invalidate-if-disabled-expected.txt: Removed.
* LayoutTests/fast/css/translate-invalidate-if-disabled.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::CSSParserContext):
* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences CSSIndividualTransformPropertiesEnabled]): Deleted.
(-[WebPreferences setCSSIndividualTransformPropertiesEnabled:]): Deleted.

Canonical link: https://commits.webkit.org/271133@main

nt1m commented on Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
```diff
@@ -3058,12 +3058,11 @@ - (void)setMediaRecorderEnabled:(BOOL)flag
 
 - (BOOL)CSSIndividualTransformPropertiesEnabled
 {
-    return [self _boolValueForKey:WebKitCSSIndividualTransformPropertiesEnabledPreferenceKey];
+    return YES;
 }
 
 - (void)setCSSIndividualTransformPropertiesEnabled:(BOOL)flag
 {
```
You should have moved this down to the `WebPrivateDeprecated` section, ditto for the WebPreferencesPrivate.h counterpart.

graouts replied:
That would be nicer, will fix in a follow-up patch. Watch this space.

graouts replied:
Filed https://github.com/WebKit/WebKit/pull/20948.
